include/pagmo/algorithms/gwo.hpp

```diff
@@ -0,0 +1,351 @@
+        }
+        // ---------------------------------------------------------------------------------------------------------
+
+        // No throws, all valid: we clear the logs
+        m_log.clear();
+
+           // Some vectors used during evolution are declared.
+        vector_double alpha_pos(dim);
+        vector_double beta_pos(dim);
+        vector_double delta_pos(dim);
+        vector_double a_vector(3); //vector coefficient for encircling prey
+        vector_double c_vector(3); //vector coefficient for encircling prey
+        vector_double d_vector(3); // position vector for alpha, beta and delta 
+        vector_double x_vector(3); // vector used to calculate position of an agent in a single dimension
+        double a; //coefficient which decrease linearly from 2 to 0 over generations
+        double r1, r2; // random coefficient between 0 and 1;
+        double alpha_score = std::numeric_limits<double>::infinity();
+        double beta_score = std::numeric_limits<double>::infinity();
+        double delta_score = std::numeric_limits<double>::infinity();
+        
+
+        // assume pop.get_x() return vector<vector_double>
+        auto  agents_position = pop.get_x();
+        
+
+
+        vector_double tmp(dim);                              // hold temporary value
+        std::uniform_real_distribution<double> drng(0., 1.); // to generate a number in [0, 1)
+
+        // initialize position of search agents
+           for(decltype(NP) j = 0u; i < NP; ++j)
+            {   
+                tmp.clear();
+                for(int i = 0; i < dim; ++i)
+            {   
+              tmp.push_back(drng(m_seed)*(ub-lb)+lb);
+            }
+                 agents_position.push_back(tmp);
+            }
+
+        // Main gwo iterations
+        for (decltype(m_gen) gen = 1u; gen <= m_gen; ++gen) 
+        {
+             // Start of the loop through each agent
+            for (decltype(NP) i = 0u; i < NP; ++i){
+                  // clip position value that goes beyond search space          
+              for(decltype(dim) k = 0; k < dim; ++k)
+                boost::algorithm::clamp(agents_position[i][k], lb, ub);
+
+
+                auto newfitness = prob.fitness(agents_position[i]);
+
+                // Update alpha, beta and delta
+                if (newfitness < alpha_score)
+                    {
+                        alpha_score = newfitness;
+                        alpha_pos = agents_position[i];
+                    }
+
+                if(newfitness > alpha_score && newfitness < beta_score){
+                    beta_score = newfitness;
+                    beta_pos = agents_position[i];
+                }
+
+                if(newfitness > alpha_score && newfitness > beta_score && newfitness <  delta_score){
+                    delta_score = newfitness;
+                    delta_pos = agents_position[i];
+
+                }
+               
+        
+            }
+             a= 2-gen*((2)/m_gen);
```

Comment:
careful here ... implicit conversions are everywhere .... 2 is int gen is unsigned m_ge too and a double .....